src/core/tsi/transport_security_grpc.h

```diff
@@ -64,6 +64,16 @@ void tsi_zero_copy_grpc_protector_destroy(tsi_zero_copy_grpc_protector* self);
 tsi_result tsi_zero_copy_grpc_protector_max_frame_size(
     tsi_zero_copy_grpc_protector* self, size_t* max_frame_size);
 
+// Reads the frame size of the input slice buffer.
```

Comment:
Can we clarify in the comments that this method does not consume any of the protected_slices and they still need to be fed into unprotect?

In general, I think we need to clarify the relationship more between this method and unprotect. For example, if we call this method once and frame_size returns 1k, then I think calling this method will continue to return 1k until we have unprotected 1k bytes. It would be good to walk through the basic CUJs like this.